api/src/db/db.service.spec.ts

```diff
@@ -26,19 +26,38 @@ describe("DbService", () => {
         expect(res.docs.length).toBe(0);
     });
 
-    it("can insert a new document", async () => {
+    it("can insert a new document and return the full document in the result's 'changes' field", async () => {
         const uuid = randomUUID();
         const doc = {
             _id: uuid,
             testData: "test123",
         };
 
-        await service.upsertDoc(doc);
+        const res = await service.upsertDoc(doc);
 
         const testGet = (await service.getDoc(uuid)) as any;
 
         expect(testGet.docs[0]._id).toBe(uuid);
         expect(testGet.docs[0].testData).toBe("test123");
+        expect(res.changes.testData).toBe("test123");
+        expect(res.changes._id).toBe(uuid);
+    });
+
+    it("can calculate a document diff and return the diff with the upsert result", async () => {
```

Comment:
Suggestion: add a third field that isn't changed and test that it isn't returned in the diff.

Reply:
Agree, added.